src/openms/source/CHEMISTRY/EnzymaticDigestionLogModel.cpp

```diff
@@ -161,9 +161,10 @@ namespace OpenMS
   {
     SignedSize count = 1;
     AASequence::ConstIterator iterator = protein.begin();
-    while (static_cast<void>(nextCleavageSite_(protein, iterator)), iterator != protein.end())
+    while (iterator != protein.end())
```

Comment:
I think this now (first count, then advance) produces different counts as before (iterator was first advanced then tested for end()).
Though I don't see for sure if it is now correct or was before.
same below.

Reply:
What du you think is best just try it out or revert back and live with the warning?

Reply:
What du you think is best just try it out or revert back and live with the warning?

Reply:
it should produce the same result with:
nextCleavageSite_(protein, iterator);
while (iterator != protein.end())
{
++count;
nextCleavageSite_(protein, iterator);
}